src/main/java/seedu/dietmanager/storage/Storage.java

```diff
@@ -151,7 +152,7 @@ public void readProfileFile() {
             Optional<Integer> age = Optional.empty();
             Optional<String> gender = Optional.empty();
             Optional<Double> height = Optional.empty();
-            Optional<Double> weight = Optional.empty();
+            Optional<List<Double>> weightList = Optional.empty();
             Optional<Double> weightGoal = Optional.empty();
```

Comment:
Can I ask why are all these optional? cause technically if these inputs are wrong, a profile should not even be generated in the first place.

Reply:
It's just an additional safeguard and to prevent the usage of null values. Otherwise, there is the values will technically "not be initialised"

Reply:
as in this storage will only store profile if the profile exist, and when profile exist, the only way for it to exist is if the user initialized the value?